diffusion/models/stable_diffusion.py

```diff
@@ -332,7 +341,9 @@ def generate(
         width: Optional[int] = None,
         num_inference_steps: Optional[int] = 50,
         guidance_scale: Optional[float] = 3.0,
-        num_images_per_prompt: int = 1,
+        rescale_guidance: bool = False,
+        rescaled_guidance_scale: float = 0.7,
```

Comment:
If you wanted to be even more concise, instead of having both the `rescale_guidance: bool = False` and `rescaled_guidance_scale: float = 0.7` parameters, you could just pass around a single `rescale_guidance: Optional[**float**] = None`. If it is None, don't apply rescale guidance - if it is set to something (e.g. 0.7), apply rescale guidance with that value.
 
 Later on in generate when you want to do the rescaling, you'd have:
 
```
# Optionally rescale the classifer free guidance
if rescale_guidance:
    std_pos = torch.std(pred_text, dim=(1, 2, 3), keepdim=True)
    std_cfg = torch.std(pred, dim=(1, 2, 3), keepdim=True)
    pred_rescaled = pred * (std_pos / std_cfg)
    pred = pred_rescaled * rescale_guidance + pred * (1 - rescale_guidance)
```

Reply:
^ Great idea.

Reply:
Thanks for the suggestion! Should be that way now.